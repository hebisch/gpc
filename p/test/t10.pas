program expon2(output);
var
        pi, spi: real;

function power(x: real; y: integer): real;
        var
                z: real;
        begin
                z := 1;
                while y>0 do
                begin
                        while not odd(y) do
                        begin
                                y := y div 2;
                                x := sqr(x);
                        end;
                        y := y-1;
                        z := x*z;
                end;
                power := z;
        end;
begin
        pi := 3.14159;
        writeln(2.0:0:10, 7:2, power(2.0, 7):15:10);
        spi := power(pi, 2);
        writeln(pi:0:10, 2:2, spi:15:10);
        writeln(spi:0:10, 2:2, power(spi, 2):15:10);
        writeln(pi:0:10, 4:2, power(pi, 4):15:10);
end.
