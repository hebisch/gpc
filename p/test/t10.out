2.0000000000 7 128.0000000000
3.1415900000 2   9.8695877281
9.8695877281 2  97.4087619227
3.1415900000 4  97.4087619227
